Source/MonoGame.Extended/Maps/Tiled/TiledMap.cs

```diff
@@ -99,16 +175,67 @@ public TiledObjectGroup GetObjectGroup(string name)
             return _objectGroups.FirstOrDefault(i => i.Name == name);
         }
         
-        public void Draw(SpriteBatch spriteBatch, Rectangle? visibleRectangle = null, GameTime gameTime = null)
+        public void Draw(SpriteBatch spriteBatch, Rectangle? visibleRectangle = null, float zoom = 1.0f, GameTime gameTime = null)
         {
+            /*
             foreach (var layer in _layers.Where(i => i.IsVisible))
                 layer.Draw(spriteBatch, visibleRectangle, gameTime: gameTime);
+            */
+
+            var rect = visibleRectangle.HasValue ? visibleRectangle.Value : Rectangle.Empty;
+
+            _worldMatrix.Translation = new Vector3(-rect.X, -rect.Y, 0);
+            _viewMatrix.Scale = new Vector3(zoom, zoom, 1.0f);
+            _basicEffect.World = _worldMatrix;
+            _basicEffect.View = _viewMatrix;
+
+            _graphicsDevice.DepthStencilState = _depthBufferState;
+            _graphicsDevice.BlendState = BlendState.AlphaBlend;
+            _graphicsDevice.SamplerStates[0] = SamplerState.PointClamp;
+
+            // Images draw
+            foreach (var pass in _basicEffect.CurrentTechnique.Passes)
+            {
+                var imageLayers = _layers.Where(layer => layer is TiledImageLayer);
```

Comment:
It's not a good idea to constantly query the layers every frame. It be better if the query result was computed and cached as a variable when the layers are initialized or when they change.